src/libmongoc/tests/test-mongoc-write-concern.c

```diff
@@ -650,21 +651,24 @@ test_write_concern_inheritance_fam_txn (void *context)
    mongoc_client_set_apm_callbacks (client, callbacks, (void *) &sent_w);
 
    wc = mongoc_write_concern_new ();
-   mongoc_write_concern_set_w (wc, 2);
+   mongoc_write_concern_set_w (wc, 1);
    mongoc_collection_set_write_concern (collection, wc);
 
    BSON_APPEND_UTF8 (&query, "firstname", "Zlatan");
    update = bson_new ();
    session_id = bson_new ();
    opts = mongoc_find_and_modify_opts_new ();
-   session = mongoc_client_start_session (client, NULL, &error);
 
-   /* conditionally append session for now */
-   if (in_txn) {
+   if (in_session) {
+      session = mongoc_client_start_session (client, NULL, &error);
+      ASSERT_OR_PRINT (session, error);
       mongoc_client_session_append (session, session_id, &error);
-      mongoc_client_session_start_transaction (session, NULL, &error);
       mongoc_find_and_modify_opts_append (opts, session_id);
    }
+   if (in_txn) {
+      BSON_ASSERT (in_session);
+      mongoc_client_session_start_transaction (session, NULL, &error);
```

Comment:
Wrap this in an `ASSERT_OR_PRINT` (or store the return in a bool) in case starting the transaction fails, so it will be easier to debug if this ever fails.